scipy/optimize/__init__.py

```diff
@@ -230,6 +230,7 @@
 
 from __future__ import division, print_function, absolute_import
 
+from .constraints import *
```

Comment:
Yes, all new submodules. The reason why there are modules without underscore prefix, is due to backward compatibility issues --- what exactly is public API was not defined for many years, so there is code out there that imports stuff from random places. The gain/harm ratio seems like it's best to leave the old modules as they are for the most part.